### PR TITLE
Replace -E with --setenv for systemd-run parameter

### PR DESCRIPTION
**Pull request by dims**
Looking at various systemd-run man pages, it looks like `-E` is not universally available while `--setenv` is. So let's switch to that please!

https://helpmanual.io/help/systemd-run/
https://manpages.ubuntu.com/manpages/xenial/man1/systemd-run.1.html

/assign @kannon92 @haircommander 
/cc @sohankunkerkar

Just ran into this for AL2 based images:
https://github.com/kubernetes-sigs/provider-aws-test-infra/pull/197

#### What type of PR is this?
/kind bug
/kind failing-test

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
```

### CODE REVIEW

**Comment by kannon92** [comment]
I pinged @sohankunkerkar to take a look.

/test

**Review by sohankunkerkar — COMMENTED** [review]
/lgtm

**Comment by kannon92** [comment]
/test pull-kubernetes-node-kubelet-serial-containerd

/hold
/lgtm

I'd like to verify that the drop in config code still works for this but otherwise I think it is good to go.

**Review by SergeyKanzhelev — APPROVED** [review]
/lgtm

I briefly looked thru the repo and it seems to be the only place where -E used

**Comment by SergeyKanzhelev** [comment]
/retest

both failures are not relevant

**Comment by dims** [comment]
/milestone v1.29

ran it by the release team [here](https://kubernetes.slack.com/archives/CJH2GBF7Y/p1700238725711589?thread_ts=1700238082.568989&cid=CJH2GBF7Y)

**Comment by kannon92** [comment]
Serial jobs look good to me. I was just making sure this doesn’t break the drop in kubelet feature. That test works

/hold cancel

**Pull request merged**
